src/lib/estimates.js

```diff
@@ -1,25 +1,56 @@
-/**
- * Calculate a weighted estimate for the interval until the next purchase
- * Current purchase a tiny bit less weight than all previous purchases
- *
- * @param {Number} lastEstimate The last estimated purchase interval
- * @param {Number} latestInterval The number of days between the most recent and previous purchases
- * @param {Number} numberOfPurchases Total number of purchases for the item
- *
- * @return {Number} Estimated number of days until the next purchase
- */
-const calculateEstimate = (lastEstimate, latestInterval, numberOfPurchases) => {
-    if (numberOfPurchases >= 2) {
-        if (isNaN(lastEstimate)) {
-            lastEstimate = 14;
-        }
-        let previousFactor = lastEstimate * numberOfPurchases;
-        let latestFactor = latestInterval * (numberOfPurchases - 1);
-        let totalDivisor = numberOfPurchases * 2 - 1;
-        return Math.round((previousFactor + latestFactor) / totalDivisor);
-    } else {
-        return latestInterval;
+// base functions
+// calculate mean
+const getMean = array => {
+  return array.reduce((acc, val) => acc + val, 0) / array.length;
+};
+// calculate standard deviation
+// we are using the calculation to find the population, rather than sample, standard deviation
+// to change to sample (if we only wanted to consider a user's more recent purchase history, for example) use array.length - 1
+const getStandardDeviation = (array, mean) => {
+  return Math.sqrt(
+    array
+      .reduce((acc, val) => acc.concat((val - mean) ** 2), [])
```

Comment:
First of all, amazing job!!  I'm so impressed with the care and thought you both put into this `estimates.js`!  I really appreciated the comments you left to help us follow the code. My ONLY suggestion for this would be changing some of the variables to be a little more descriptive so that it's easier to understand quickly. For example, instead of "acc" maybe "squaredArray" or something like that.

Reply:
Yes, this commenting is top-notch.

Reply:
Yes! Great call on descriptive variables.

Reply:
Yes! I'll update for better readability.